pkg/config/autodiscover/autodiscover_generic.go

```diff
@@ -0,0 +1,110 @@
+// Copyright (C) 2020 Red Hat, Inc.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+package autodiscover
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/test-network-function/test-network-function/pkg/config/configsections"
+)
+
+const (
+	genericLabelName = "generic"
+	underTestValue   = "target"
+	// partnerLabelValue = "partner"
+	orchestratorValue          = "orchestrator"
+	fsDiffMasterValue          = "fs_diff_master"
+	skipConnectivityTestsLabel = "skip_connectivity_tests"
+)
+
+// BuildGenericConfig builds a `configsections.TestConfiguration` from the current state of the cluster,
+// using labels and annotations to populate the data.
+func BuildGenericConfig() (conf configsections.TestConfiguration) {
+	var partnerContainers []configsections.Container // PartnerContainers is built from all non-target containers
+
+	// an orchestrator must be identified
+	orchestrator, err := getContainerByLabel(genericLabelName, orchestratorValue)
+	if err != nil {
+		log.Fatalf("failed to identify a single test orchestrator container: %s", err)
+	}
+	partnerContainers = append(partnerContainers, orchestrator)
+	conf.TestOrchestrator = orchestrator.ContainerIdentifier
+
+	// there must be containers to test
+	containersUnderTest, err := getContainersByLabel(genericLabelName, underTestValue)
+	if err != nil {
+		log.Fatalf("found no containers to test: %s", err)
+	}
+	conf.ContainersUnderTest = containersUnderTest
+
+	// the FS Diff master container is optional
+	fsDiffMasterContainer, err := getContainerByLabel(genericLabelName, fsDiffMasterValue)
+	if err == nil {
+		partnerContainers = append(partnerContainers, fsDiffMasterContainer)
+		conf.FsDiffMasterContainer = fsDiffMasterContainer.ContainerIdentifier
+	} else {
+		log.Warnf("an error (%s) occurred when getting the FS Diff Master Container. Attempting to continue", err)
```

Comment:
The change to give warnings for the FS Diff Master and the connectivity test exclusions was deliberate. Running with no FS Diff master configured is a sensible setup and will simply result in that test not being run, as with the case if no containers are excluded.
The use of a warning is to show that nothing was found in the potential case when that was caused by a misconfiguration rather than that config having been deliberately unset. Running with no test partner is not so possible, hence the exit.

Reply:
I just meant the error string.  No worries though.